modules/programs/msmtp.nix

```diff
@@ -0,0 +1,77 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.msmtp;
+
+  dag = config.lib.dag;
+
+  msmtpAccounts = filter (a: a.msmtp.enable)
+    (attrValues config.accounts.email.accounts);
+
+  onOff = p: if p then "on" else "off";
+
+  accountStr = account: with account;
+    concatStringsSep "\n" (
+      [ "account ${name}" ]
+      ++ mapAttrsToList (n: v: n + " " + v) (
+        {
+          host = smtp.host;
+          from = address;
+          auth = "on";
+          user = userName;
+          tls = onOff smtp.tls.enable;
+          tls_starttls = onOff smtp.tls.useStartTls;
+          tls_trust_file = smtp.tls.certificatesFile;
+        }
+        // optionalAttrs (smtp.port != null) {
+          port = toString smtp.port;
+        }
+        // optionalAttrs (passwordCommand != null) {
+          # msmtp requires the password to finish with a newline.
+          passwordeval = ''${pkgs.bash}/bin/bash -c "${toString passwordCommand}; echo"'';
+        }
+      )
+      ++ optional primary "\naccount default : ${name}"
+    );
+
+  configFile = mailAccounts: ''
+    # Generated by Home Manager.
+
+    ${cfg.extraConfig}
+
+    ${concatStringsSep "\n\n" (map accountStr mailAccounts)}
+  '';
+
+in
+
+{
+
+  options = {
+    programs.msmtp = {
+      enable = mkEnableOption "msmtp";
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = "";
+        description = ''
+          Extra configuration lines to add to <filename>~/.msmtprc</filename>.
+          See <link xlink:href="https://marlam.de/msmtp/msmtprc.txt"/> for examples.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.msmtp ];
+
+    home.file.".msmtprc".text = configFile msmtpAccounts;
```

Comment:
I miss the pseudo xdg I had built around but that was opiniated :)

Reply:
arf since 1.6.7, msmtp supports xdg so writing it in xdg.configHome/msmtp/config might be best

Reply:
Yeah, I updated msmtp to 1.6.8 in Nixpkgs master. It should be in NixOS 18.09 and once that is released I'd be happy to move this to the XDG config directory. For now I think it is best to keep at `~/.msmtprc` to let the plain `msmtp` command work without extra command line arguments.